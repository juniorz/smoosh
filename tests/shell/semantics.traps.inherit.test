( 
	trap "echo got SIGINT" INT

        # default trap will mean we actually get a QUIT, overriding the default on asyncs
	trap - QUIT

	mypid=$($TEST_SHELL -c 'echo $PPID')

        # this can be overridden
	kill -s INT "$mypid" || exit 4

        # will kill this shell, 
	echo "sending SIGQUIT"
	kill -s QUIT "$mypid" || exit 2
	exit 0
) &
wait $!
echo $?

