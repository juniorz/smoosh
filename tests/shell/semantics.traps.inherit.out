got SIGINT
sending SIGQUIT
131
